src/main/java/com/rackspacecloud/metrics/rollup/domain/Metric.java

```diff
@@ -1,118 +1,151 @@
 package com.rackspacecloud.metrics.rollup.domain;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Data;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
+@Data
 public class Metric implements IReducer<Metric> {
-    public String rollupKey;
 
-    public Map<String, List<Long>> iValuesForRollup;
-    public Map<String, List<Double>> fValuesForRollup;
+    private static final Logger LOGGER = LoggerFactory.getLogger(Metric.class);
+
+    public String rollupKey;
 
-    public String timestamp;
+//    public Map<String, List<Long>> iValuesForRollup;
+//    public Map<String, List<Double>> fValuesForRollup;
 
     // InfluxDB tags
-    public AccountType accountType;
-    public String account;
-    public String device;
-    public String deviceLabel;
-    public Map<String, String> deviceMetadata;
-    public MonitoringSystem monitoringSystem;
-    public Map<String, String> systemMetadata;
-    public String collectionLabel;
-    public String collectionTarget;
-    public Map<String, String> collectionMetadata;
+    String accountType;
+    String account;
+    String device;
+    String deviceLabel;
+    Map<String, String> deviceMetadata;
+    String monitoringSystem;
+    Map<String, String> systemMetadata;
+    String collectionName;
+    String collectionLabel;
+    String collectionTarget;
+    Map<String, String> collectionMetadata;
 
     // InfluxDB fields
     public Map<String, Long> ivalues;
     public Map<String, Double> fvalues;
     public Map<String, String> svalues;
     public Map<String, String> units;
 
+
+
+    public Map<String, RollupBucket<Long>> rolledUpIValues;   // Keeps rolled up ivalue
+    public Map<String, RollupBucket<Double>> rolledUpFValues; // Keeps rolled up fvalue
+
+    @Data
+    public static class RollupBucket<T extends Number> {
+        private T min;
+        private T max;
+        private T mean;
```

Comment:
Not related to this PR, but it would be easy and cheap to add quantiles (median, 1%, 99%, etc) to this using tdigest, see https://github.com/tdunning/t-digest